contrib/hack/start-server.sh

```diff
@@ -23,20 +23,36 @@ export PATH=${ROOT}/contrib/hack:${PATH}
 # Clean up old containers if still around
 docker rm -f etcd apiserver > /dev/null 2>&1 || true
 
-# Start etcd, our DB
+# Start etcd, our DB. Also, map 6443 (api server port) to some random
+# port on the host - then ask Docker for the port # as we'll need to use
+# that when we talk to it.
 echo Starting etcd
-docker run -ti --name etcd -d --net host quay.io/coreos/etcd > /dev/null
+docker run -ti --name etcd -d -p 6443 quay.io/coreos/etcd > /dev/null
+PORT=$(docker port etcd 6443 | sed "s/.*://")
 
 # And now our API Server
 echo Starting the API Server
-docker run -ti --name apiserver -d --net host \
-	-v ${ROOT}:/go/src/github.com/kubernetes-incubator/service-catalog \
-	-v ${ROOT}/.var/run/kubernetes-service-catalog:/var/run/kubernetes-service-catalog \
-	-v ${ROOT}/.kube:/root/.kube \
-	scbuildimage \
-	bin/apiserver -v 10 --etcd-servers http://localhost:2379 > /dev/null
+docker run -tid --name apiserver \
+    -v ${ROOT}:/go/src/github.com/kubernetes-incubator/service-catalog \
+    -v ${ROOT}/.var/run/kubernetes-service-catalog:/var/run/kubernetes-service-catalog \
+    -v ${ROOT}/.kube:/root/.kube \
+    -e KUBERNETES_SERVICE_HOST=localhost \
+    -e KUBERNETES_SERVICE_PORT=6443 \
+    --net container:etcd \
```

Comment:
I like that we use a specific container. Is the apiserver still accessible from outside the container?